mixnet/node/src/lib.rs

```diff
@@ -92,16 +90,14 @@ impl MixnetNode {
     }
 
     async fn handle_final_payload(
-        mut socket: TcpStream,
         _private_key: PrivateKey,
         client_address: SocketAddr,
+        payload: Payload,
     ) -> Result<(), Box<dyn Error>> {
         // TODO: Reuse a conn instead of establishing ad-hoc conns
         let mut client_stream = TcpStream::connect(client_address).await?;
-
         // TODO: Decrypt the final payload using the private key
-        let _ = tokio::io::copy(&mut socket, &mut client_stream).await?;
-
+        client_stream.write_all(payload.as_bytes()).await?;
```

Comment:
@youngjoon-lee This is the only thing im not so happy about. As your code was a better option. The thing is that we have this method that does exactly the same as one of the `forward` ones. We may be able to generalize over streams if we do not need to read the payload (having a `FinalPayload(Box<Stream<u8>>)` instead of the payload).
What do you think?

Reply:
I gave it a try. You can see the differences between both commits.

Reply:
I love the second commit!